homeassistant/components/device_tracker/owntracks.py

```diff
@@ -297,30 +324,28 @@ def _async_transition_message_leave(hass, context, message, location):
     if location in regions:
         regions.remove(location)
 
-    new_region = regions[-1] if regions else None
-
-    if new_region:
-        # Exit to previous region
-        zone = hass.states.get(
-            "zone.{}".format(slugify(new_region)))
-        _set_gps_from_zone(kwargs, new_region, zone)
-        _LOGGER.info("Exit to %s", new_region)
-        yield from context.async_see(**kwargs)
-        yield from context.async_see_beacons(dev_id, kwargs)
-        return
-
-    else:
-        _LOGGER.info("Exit to GPS")
-
-        # Check for GPS accuracy
-        if context.async_valid_accuracy(message):
-            yield from context.async_see(**kwargs)
-            yield from context.async_see_beacons(dev_id, kwargs)
-
     beacons = context.mobile_beacons_active[dev_id]
     if location in beacons:
         beacons.remove(location)
         _LOGGER.info("Remove beacon %s", location)
+        yield from context.async_see_beacons(hass, dev_id, kwargs)
+    else:
```

Comment:
Instead of this `else`, let's add a return to the above `if`. That way we don't get in if…else spaghetti

Reply:
Sounds good.